internal/surfer/gcloud/builder.go

```diff
@@ -0,0 +1,403 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/googleapis/librarian/internal/sidekick/api"
+	"github.com/googleapis/librarian/internal/surfer/gcloud/utils"
+	"github.com/iancoleman/strcase"
+)
+
+// NewCommand constructs a single gcloud command definition from an API method.
+// This function assembles all the necessary pieces: help text, arguments,
+// request details, and async configuration.
+func NewCommand(method *api.Method, overrides *Config, model *api.API, service *api.Service) (*Command, error) {
+	// We look up the help text and API definition for this specific method in the
+	// `gcloud.yaml` configuration file.
+	rule := findHelpTextRule(method, overrides)
+
+	// We initialize the command with some default values.
+	cmd := &Command{
+		AutoGenerated: true,
+	}
+
+	if len(overrides.APIs) > 0 {
+		cmd.Hidden = overrides.APIs[0].RootIsHidden
+	} else {
+		// Default to hidden if no API overrides are provided.
+		cmd.Hidden = true
+	}
+
+	// If a help text rule was found in the config, we apply it to the command.
+	if rule != nil {
+		cmd.HelpText = HelpText{
+			Brief:       rule.HelpText.Brief,
+			Description: rule.HelpText.Description,
+			Examples:    strings.Join(rule.HelpText.Examples, "\n\n"),
+		}
+	}
+
+	// Infer default release track from proto package.
+	// TODO(https://github.com/googleapis/librarian/issues/3289): Allow gcloud config to overwrite the track for this command.
+	inferredTrack := utils.InferTrackFromPackage(method.Service.Package)
+	cmd.ReleaseTracks = []string{strings.ToUpper(inferredTrack)}
+
+	// The core of the command generation happens here: we generate the arguments,
+	// request details, and async configuration.
+	args, err := newArguments(method, overrides, model, service)
+	if err != nil {
+		return nil, err
+	}
+	cmd.Arguments = args
+	cmd.Request = newRequest(method, overrides, model)
+	if method.OperationInfo != nil {
+		cmd.Async = newAsync(method, overrides)
+	}
+
+	return cmd, nil
+}
+
+// newArguments generates the set of arguments for a command by parsing the
+// fields of the method's request message.
+func newArguments(method *api.Method, overrides *Config, model *api.API, service *api.Service) (Arguments, error) {
+	args := Arguments{}
+	if method.InputType == nil {
+		return args, nil
+	}
+
+	// We iterate over each field in the method's request message (e.g., `CreateInstanceRequest`).
+	for _, field := range method.InputType.Fields {
+		// The "parent" field is a special case. Its information is captured by the
+		// primary resource argument, so we skip it here to avoid creating a redundant flag.
```

Comment:
Not totally sure I follow this logic. Why is `name` not also redundant?

Reply:
Since name is the primary resource and is handled differently for Get and Delete methods, we don't skip it initially, though we do skip it later. However, you are right; this logic is confusing. I've created an issue to track the refactoring: https://github.com/googleapis/librarian/issues/3412. We might need to split this into different segments, but we can continue that discussion within the issue.